cmd/packet_analyzer/capture.go

```diff
@@ -27,30 +31,87 @@ type PacketRequest struct {
 	Destination string
 	// The contents of the packet.
 	Contents []int
+	// Packet timestamp
+	Timestamp time.Time
 }
 
 var (
 	// Mapping of server names to channels of PacketRequests acting as queues.
 	packetChannels = make(map[string]chan *PacketRequest)
 	// Mapping of server names to the ordered packets.
 	packetQueues = make(map[string][]Packet)
+	//go:embed "templates/index.html"
+	uiTemplate string
 )
 
 // startCapturing spins up an HTTP handler to await packet submissions from one
 // or more running servers. On exit it will write the contents of each session
 // to a file for you to do what you will.
-func startCapturing(serverAddr, folder string, httpPort, tcpPort int) {
+func startCapturing(serverAddr, folder string, httpPort, tcpPort, managePort int, auto bool) {
 	// Register a signal handler to dump the packet lists before exiting.
 	signalChan := make(chan os.Signal, 1)
 	signal.Notify(signalChan, syscall.SIGINT, syscall.SIGCONT, syscall.SIGTERM)
-	go captureExitHandler(signalChan, folder)
+	go captureExitHandler(signalChan, folder, auto)
 
 	go listenForTCPPackets(serverAddr, tcpPort)
+	if managePort != 0 {
+		go startManageServer(serverAddr, managePort)
+	}
 	listenForHTTPPackets(serverAddr, httpPort)
 }
 
+func startManageServer(serverAddr string, managePort int) {
+	addr := fmt.Sprintf("%s:%d", serverAddr, managePort)
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", manage)
+	mux.HandleFunc("/ui", ui)
```

Comment:
Now that I've played with this a little more, from a purely API/webapp perspective it would probably be cleaner if it looked like this:
```
/ui -> page with some crappy links to the "tools" (sort of like pprof)
/ui/diff -> the diffing tool
/ -> list the sessions (as you currently have)
/<session-name> -> the packet view that's currently obtained with the query arg
```
However this is a suggestion and for now this ought to work fine, so I'm going to leave this to your discretion

Reply:
I will probably change routing in future once there are really some tools, guess it's ok to leave as is for now